mkosi/__init__.py

```diff
@@ -6412,8 +6415,8 @@ def load_args(args: argparse.Namespace) -> MkosiConfig:
             args.mirror = "http://deb.debian.org/debian"
         elif args.distribution == Distribution.ubuntu:
             args.mirror = "http://archive.ubuntu.com/ubuntu"
-            if platform.machine() == "aarch64":
-                args.mirror = "http://ports.ubuntu.com/"
+            if platform.machine() == "aarch64" or args.architecture == "aarch64":
+                args.mirror = "http://ports.ubuntu.com/ubuntu-ports"
```

Comment:
Same here, please remove the `platform.machine()` check and keep only the `args.architecture` check